### ISSUE TITLE
RTL8821AU with 5ghz support (TPLINK Archer T2U Nano and others)

### ISSUE THREAD

**Issue by valerino** [MiSTer-devel/Linux-Kernel_MiSTer#13]
hi,
i've tried the TPLink Archer T2U Nano with Mister (latest update) and it works flawlessy under 2.4ghz. With 5ghz networks, it doesn't work at all (can't find the network).
after a bit of search around, seems to be a common problem and i managed to compile an updated version of the driver, with specific Makefile tweaks for the Mister.

[this](https://github.com/morrownr/8821au-20210708) is the original repo and [here is the specific MiSTer build info](https://github.com/valerino/8821au-20210708/blob/main/README_mister.md) on my fork.

it works flawlessy then with both 2.4 and 5ghz networks.

should be included in the main kernel replacing the old driver, imho ....

regards!

**Comment by sorgelig** [comment]
are you talking about this fix?
https://github.com/valerino/8821au-20210708/commit/105643b7db7f9caa403c153ec5c0df6b13d77423

did you try to modify it in included driver to make it work with 5ghz?

**Comment by valerino** [comment]
hi! sincerely i didn't tried to mess with the default driver. i just made that updated repo work as a loadable module (just makefile fixes as you can see). should be safe to use the updated repo in the mister kernel, i think.

**Comment by sorgelig** [comment]
it's not as straight as you might think. There were additions to support different dongles. If i replace it as is, then will get tons of complains that some dongle with the same chip stopped to work. That's why it's much better to apply patch to solve specific problem than replace a whole driver.

**Comment by valerino** [comment]
ok, got it. so just take a note for the next release to include the patch. you can close this, thanks for the replies!

**Comment by sorgelig** [comment]
But did you check if that fix really fixes your issue?

**Comment by valerino** [comment]
oh sure, i can do it. is there a quick guide on  how to recompile the kernel ?

EDIT: nevermind, found out .... wait a bit, and i'll tell you if it works.

**Comment by valerino** [comment]
ok, no ... seems the patch doesn't work. still see only 2.4ghz. probably something else changed :(

**Comment by sorgelig** [comment]
i've tested my TPLINK Archer T2U Nano.
It works fine with 5GHz with current driver. I also have D-Link DWA-171 which is 8821au as well and it works fine with 5GHz too.
Most likely you have bad signal. Such tiny wifi dongles have not good antenna and require very good signal.

Btw, your driver is 8821au-only while current driver is 8821ua/8812au - more dongles are supported.

**Comment by valerino** [comment]
uhm .... unlikely, i have the router next to my pc. anyway, ok .... i'm keeping this for my private setup only so. thanks for investigating, though!

**Comment by movisman** [comment]
Hi,

I just wanted to confirm for @valerino that I had a T2U nano for a short while, and had the same issue - only 2.4ghz networks would be seen. 5ghz was non functional. Running "iwlist freq" would only return the 2.4ghz channels. No signal issue my end - put my ASUS AC53 Nano in (RTL8822BU based) and 5ghz networks are seen once again.

I tried the same T2U nano on Windows 10, and both 2.4 and 5ghz networks are seen just fine. But on my MiSTer, only 2.4ghz was operational. Someone on discord kindly tested theirs and 5ghz was fine for them (like Sorgelig), but they had a US variant. Same hardware revision. Mine was UK - I wondered if it was related to country/regulatory stuff in the driver which was restricting 5ghz usage. I have no idea to be honest.

Unfortunately I no longer have the device as I returned it to Amazon.

But just wanted to let you know I had the same thing. If you want to use 5ghz, I can confirm the ASUS AC53 Nano works for me.
